src/QueryM.purs

```diff
@@ -431,15 +428,12 @@ txOutRefToTransactionInput { txId, index } = do
 -- hexToByteArray for now.
 -- | Converts an Ogmios TxOut to (internal) TransactionOutput
 ogmiosTxOutToTransactionOutput
-  :: OgmiosTxOut
-  -> Effect (Maybe Transaction.TransactionOutput)
+  :: JsonWsp.OgmiosTxOut
+  -> Maybe Transaction.TransactionOutput
 ogmiosTxOutToTransactionOutput { address: address'', value, datum } = do
   address' <- ogmiosAddressToAddress address''
-  pure $ case address' of
-    Nothing -> Nothing
-    Just address ->
-      pure $ wrap
-        { address
-        , amount: value
-        , data_hash: datum >>= hexToByteArray <#> wrap
-        }
+  pure $ wrap
+    { address: address'
+    , amount: value
+    , data_hash: datum >>= hexToByteArray <#> wrap
+    }
```

Comment:
This is a relief. We could probably remove the `address''` completely with the new functionality and start with `{ address, value, datum }`, then
` address' <- ogmiosAddressToAddress address`